packages/caravan-fees/src/transactionAnalyzer.ts

```diff
@@ -0,0 +1,539 @@
+   */
+  get targetFeeRate(): number {
+    return this._targetFeeRate;
+  }
+
+  /**
+   * Calculates and returns the minimum fee rate required for a successful RBF.
+   * @returns {string} The minimum RBF fee rate in satoshis per vbyte
+   */
+  get rfbFeeRate(): string {
+    if (!this._rbfFeeRate) {
+      const currentFeeRate = this.feeRate;
+
+      const minReplacementFeeRate = BigNumber.max(
+        new BigNumber(currentFeeRate).plus(this.incrementalRelayFee),
+        new BigNumber(this.targetFeeRate),
+      );
+
+      this._rbfFeeRate = minReplacementFeeRate.toString();
+    }
+    return this._rbfFeeRate;
+  }
+
+  /**
+   * Calculates and returns the fee rate required for a successful CPFP.
+   * @returns {string} The CPFP fee rate in satoshis per vbyte
+   */
+  get cpfpFeeRate(): string {
+    if (!this._cpfpFeeRate) {
+      const originalTxVBytes = this.vsize;
+      const packageSize = originalTxVBytes + ESTIMATED_CHILD_TX_SIZE;
+      const desiredPackageFee = new BigNumber(this.targetFeeRate).multipliedBy(
+        packageSize,
+      );
+      const expectedFeeRate = BigNumber.max(
+        desiredPackageFee.minus(this.fee),
+        new BigNumber(0),
+      ).dividedBy(ESTIMATED_CHILD_TX_SIZE);
+
+      this._cpfpFeeRate = expectedFeeRate.toString();
+    }
+    return this._cpfpFeeRate;
+  }
+
+  /**
+   * Calculates and returns the recommended fee rate based on the current transaction state and target fee rate.
+   * @returns {number} The recommended fee rate in satoshis per vbyte
+   */
+  get getRecommendedFeeRate(): number {
+    const currentFeeRate = parseFloat(this.feeRate);
+    const targetFeeRate = this.targetFeeRate;
+
+    if (currentFeeRate >= targetFeeRate) {
+      return currentFeeRate; // Current fee rate is already sufficient
+    }
+
+    if (this.canRBF && (!this.canCPFP || this.rfbFeeRate <= this.cpfpFeeRate)) {
+      return Math.max(Number(this.rfbFeeRate), targetFeeRate);
+    } else if (this.canCPFP) {
+      return Math.max(Number(this.cpfpFeeRate), targetFeeRate);
+    } else {
+      return targetFeeRate; // If neither RBF nor CPFP is possible, return the target fee rate
+    }
+  }
+
+  /**
+   * Gets the dust threshold used for analysis.
+   * @returns {string} The dust threshold in satoshis
+   */
+  get getDustThreshold(): string {
+    return this.dustThreshold.toString();
+  }
+
+  /**
+   * Sets a new target fee rate for the transaction analysis.
+   * @param {number} rate - The new target fee rate in satoshis per vbyte
+   */
+  set setTargetFeeRate(rate: number) {
+    this._targetFeeRate = rate;
+    this._recommendedStrategy = null; // Reset cached value
+  }
+
+  /**
+   * Sets the index of the change output in the transaction.
+   * @param {number | undefined} index - The index of the change output, or undefined if no change output
+   */
+  set setChangeOutput(index: number | undefined) {
+    this.changeOutputIndex = index;
+    this._outputs = null; // Reset cached value
```

Comment:
why reset outputs?

Reply:
That was a logical error ... It is corrected in new class (also this method is no longer there)